src/coreclr/jit/gentree.cpp

```diff
@@ -24881,6 +25484,13 @@ GenTree* Compiler::gtNewSimdShuffleNode(
     var_types simdBaseType = JitType2PreciseVarType(simdBaseJitType);
     assert(varTypeIsArithmetic(simdBaseType));
 
+    if (isUnsafe)
+    {
+        // For ShuffleUnsafe, delegate to the variable implementation to get the same behaviour for
+        // ShuffleUnsafe with constant vs variable indices for free.
+        return gtNewSimdShuffleNodeVariable(type, op1, op2, simdBaseJitType, simdSize, isUnsafe);
+    }
+
```

Comment:
@tannergooding this change you requested causes `ShuffleUnsafe(constant, constant)` to not be a constant - e.g.,
```csharp
public void ShuffleUnsafeLong128() => Consume(Vector128.ShuffleUnsafe(Vector128.Create(0, 1), Vector128.Create(1, 0)));
```
-->
```asm
; Assembly listing for method ShuffleUnsafeBenchSse4Only:ShuffleUnsafeLong128():this (Tier1)
; Emitting BLENDED_CODE for X64 with AVX - Apple
; Tier1 code
; optimized code
; optimized using Synthesized PGO
; rsp based frame
; partially interruptible
; with Synthesized PGO: fgCalledCount is 100
; No PGO data

G_M000_IG01:                ;; offset=0x0000
       sub      rsp, 24
 
G_M000_IG02:                ;; offset=0x0004
       vmovups  xmm0, xmmword ptr [reloc @RWD00]
       vmovups  xmm1, xmmword ptr [reloc @RWD16]
       vpermilps xmm1, xmm1, xmmword ptr [reloc @RWD32]
       vpor     xmm1, xmm1, xmmword ptr [reloc @RWD48]
       vpermilps xmm0, xmm0, xmm1
       vmovups  xmmword ptr [rsp], xmm0
       call     [ShuffleUnsafeBenchSse4Only:Consume[System.Runtime.Intrinsics.Vector128`1[long]](System.Runtime.Intrinsics.Vector128`1[long])]
       nop      
 
G_M000_IG03:                ;; offset=0x0036
       add      rsp, 24
       ret      
 
RWD00  	dq	0000000000000000h, 0000000000000001h
RWD16  	dq	0000000000000002h, 0000000000000000h
RWD32  	dq	0000000000000000h, 0000000200000002h
RWD48  	dq	0000000100000000h, 0000000100000000h

; Total bytes of code 59
```

vs

```csharp
public void ShuffleLong128() => Consume(Vector128.Shuffle(Vector128.Create(0, 1), Vector128.Create(1, 0)));
```
-->
```asm
; Assembly listing for method ShuffleBenchSse4Only:ShuffleLong128():this (Tier1)
; Emitting BLENDED_CODE for X64 with AVX - Apple
; Tier1 code
; optimized code
; optimized using Synthesized PGO
; rsp based frame
; partially interruptible
; with Synthesized PGO: fgCalledCount is 100
; No PGO data

G_M000_IG01:                ;; offset=0x0000
       sub      rsp, 24
 
G_M000_IG02:                ;; offset=0x0004
       vpermilpd xmm0, xmmword ptr [reloc @RWD00], 1
       vmovups  xmmword ptr [rsp], xmm0
       call     [ShuffleBenchSse4Only:Consume[System.Runtime.Intrinsics.Vector128`1[long]](System.Runtime.Intrinsics.Vector128`1[long])]
       nop      
 
G_M000_IG03:                ;; offset=0x001A
       add      rsp, 24
       ret      
 
RWD00  	dq	0000000000000000h, 0000000000000001h
```

How should I fix this? Should it just treat it as `Shuffle` if we don't have any out of range indices instead? Or is there another approach I should implement?

Reply:
I've implemented the above idea - let me know if you prefer it solved a different way by un-resolving and commenting :)